libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -265,6 +265,7 @@ assembly::Expression Parser::parseExpression()
 		return parseCall(std::move(operation));
 	else if (operation.type() == typeid(Instruction))
 	{
+		solAssert(m_flavour == AsmFlavour::Loose, "");
```

Comment:
I think instead of an assert, this should be a proper `fatalParserError`, since this seems to be the right place to show that a non-functional instruction was parsed in strict mode.

Reply:
Together with the checks in line 255, we should be in the situation that the next token is a `(` and thus the `if` in line 264 will be active, and not this else case.